### PR TITLE
Modify diffsender to accept diff in byte-array.

### PR DESCRIPTION
**Pull request by samutamm**
More tests and refactoring to courseSubmitter.

### CODE REVIEW

File: config.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #Updated properties
-#Thu Jun 04 11:51:31 EEST 2015
-serverAddress=http\://tmc.mooc.fi/staging
-serverPort=36319
```
Comment (Coolnesss) [inline config.properties]: Get rid of this file

File: src/test/java/hy/tmc/cli/spyware/DiffSenderTest.java [hunk 2]
```diff
@@ -74,8 +75,31 @@ public void testSendToAllUrls() throws IOException {
...
+    public void testSendToAllUrlsWithByteArray() throws IOException {
+        final File file = new File("testResources/test.zip");
+        byte[] byteArray = Files.toByteArray(file);
+        Course testCourse = new Course();
+        testCourse.setSpywareUrls(
+                Arrays.asList(new String[]{spywareUrl})
+        );
+        List<HttpResult> results = sender.sendToSpyware(byteArray, testCourse);
+        for (HttpResult res : results) {
+            assertEquals(200, res.getStatusCode());
```
Comment (Coolnesss) [inline src/test/java/hy/tmc/cli/spyware/DiffSenderTest.java]: All tests only test 200 success

File: src/test/java/hy/tmc/cli/spyware/DiffSenderTest.java [hunk 3]
```diff
@@ -86,7 +124,6 @@ private void startWiremock() {
                 .withHeader("X-Tmc-Version", equalTo("1"))
                 .withHeader("X-Tmc-Username", equalTo(ClientData.getUsername()))
                 .withHeader("X-Tmc-Password", equalTo(ClientData.getPassword()))
-                .withRequestBody(containing("test.zip"))
                 .willReturn(
                         aResponse()
                                 .withBody("OK")
```
Comment (jamox) [inline src/test/java/hy/tmc/cli/spyware/DiffSenderTest.java]: Why this is not just one line -- it is way less than 100 chars and the oneline version is still readable

Comment (samutamm) [inline src/test/java/hy/tmc/cli/spyware/DiffSenderTest.java]: done

**Pull request merged**